From 125bd387db5250ae1af19de6720abac375e21f98 Mon Sep 17 00:00:00 2001
From: Alejandro Mery <amery@geeks.cl>
Date: Sun, 14 Oct 2012 15:44:15 +0200
Subject: [PATCH 800/944] wireless: rtl8192cu: fix problem because of
 wakelocks api change

by disabling some weird sanity checks that assume too much
---
 drivers/net/wireless/rtl8192cu/os_dep/osdep_service.c | 13 +++++++++----
 1 file changed, 9 insertions(+), 4 deletions(-)

diff --git a/drivers/net/wireless/rtl8192cu/os_dep/osdep_service.c b/drivers/net/wireless/rtl8192cu/os_dep/osdep_service.c
index 6e11379..4b84121 100644
--- a/drivers/net/wireless/rtl8192cu/os_dep/osdep_service.c
+++ b/drivers/net/wireless/rtl8192cu/os_dep/osdep_service.c
@@ -1016,6 +1016,7 @@ inline void rtw_suspend_lock_init()
 inline void rtw_suspend_lock_uninit()
 {
 
+#if 0
 	#if  defined(CONFIG_WAKELOCK) || defined(CONFIG_ANDROID_POWER)
 	DBG_871X("##########%s###########\n", __FUNCTION__);
 	if(rtw_suspend_lock.link.next == LIST_POISON1 || rtw_suspend_lock.link.prev == LIST_POISON2) {
@@ -1023,7 +1024,8 @@ inline void rtw_suspend_lock_uninit()
 		return;	
 	}
 	#endif
-	
+#endif
+
 	#ifdef CONFIG_WAKELOCK
 	wake_lock_destroy(&rtw_suspend_lock);
 	#elif defined(CONFIG_ANDROID_POWER)
@@ -1034,7 +1036,7 @@ inline void rtw_suspend_lock_uninit()
 
 inline void rtw_lock_suspend()
 {
-
+#if 0
 	#if  defined(CONFIG_WAKELOCK) || defined(CONFIG_ANDROID_POWER)
 	//DBG_871X("##########%s###########\n", __FUNCTION__);
 	if(rtw_suspend_lock.link.next == LIST_POISON1 || rtw_suspend_lock.link.prev == LIST_POISON2) {
@@ -1042,7 +1044,8 @@ inline void rtw_lock_suspend()
 		return;	
 	}
 	#endif
-	
+#endif
+
 	#ifdef CONFIG_WAKELOCK
 	wake_lock(&rtw_suspend_lock);
 	#elif defined(CONFIG_ANDROID_POWER)
@@ -1052,6 +1055,7 @@ inline void rtw_lock_suspend()
 
 inline void rtw_unlock_suspend()
 {
+#if 0
 	#if  defined(CONFIG_WAKELOCK) || defined(CONFIG_ANDROID_POWER)
 	//DBG_871X("##########%s###########\n", __FUNCTION__);
 	if(rtw_suspend_lock.link.next == LIST_POISON1 || rtw_suspend_lock.link.prev == LIST_POISON2) {
@@ -1059,7 +1063,8 @@ inline void rtw_unlock_suspend()
 		return;	
 	}
 	#endif
-	
+#endif
+
 	#ifdef CONFIG_WAKELOCK
 	wake_unlock(&rtw_suspend_lock);
 	#elif defined(CONFIG_ANDROID_POWER)
-- 
1.8.0

